Source/Utility/wavetables.h

```diff
@@ -0,0 +1,390 @@
+#pragma once
+#include <math.h>
+#include <array>
+#include <vector>
+
+#include "dsp.h"
+
+namespace daisysp
+{
+
+// NOTE: Using (size + 1) to fix the computational complexity with table wrapping
+// i.e. data[i], data[i +1] where i == size
+struct WaveBuffer
+{
+    static const uint16_t wt_size  = 2048;
+    float                 top_freq = 0;
+    float                 data[wt_size + 1]{0};
+};
+
+/**
+ * @brief Abstraction around band-limited wavetables
+ */
+struct WaveTable
+{
+    WaveBuffer *buff = nullptr;
+
+    float GetSample(float id)
+    {
+        // return interpolate_ ? Interpolate(id) : _get(static_cast<uint16_t>(id));
+        return Interpolate(id);
+    }
+
+    /**
+     * @brief Selects underlying waveform table matching the given normalized frequency.
+     * 
+     * @param norm_freq normalized frequency (frequency * 1 / sample_rate)
+     */
+    void SetTopFreq(float norm_freq)
+    {
+        if(buffers.size() == 1)
+        {
+            return;
+        }
+
+        size_t curr_buffer_ = 0;
+        while(curr_buffer_ < buffers.size()
+              && norm_freq >= buffers[curr_buffer_]->top_freq)
+        {
+            ++curr_buffer_;
+        }
+
+        curr_buffer = curr_buffer_;
+        buff        = buffers[curr_buffer];
+    }
+
+    WaveBuffer *Allocate(WaveBuffer *buff_)
+    {
+        buff = buff_;
+        buffers.push_back(buff);
+        buff->top_freq = 0;
+        return buffers.back();
+    }
+
+  private:
+    float _get(size_t idx) { return buffers[curr_buffer]->data[idx]; }
+
+    float Interpolate(float frame)
+    {
+        auto  intPart  = static_cast<uint16_t>(frame);
+        float fracPart = frame - static_cast<float>(intPart);
+        float samp0    = _get(intPart);
+        float samp1    = _get(intPart + 1);
+        return samp0 + (samp1 - samp0) * fracPart;
+    }
+
+    size_t curr_buffer = 0;
+
+    std::vector<WaveBuffer *> buffers;
+};
+
+struct Tables
```

Comment:
As suggested by @antisvin in the forum, it might be a good idea to provide a templated FFT implementation instead of one hardcoded below.